tests/unit/plugins/test_base_python.py

```diff
@@ -0,0 +1,169 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright 2024 Canonical Ltd.
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License version 3 as published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+from pathlib import Path
+from textwrap import dedent
+from typing import Literal
+
+import pytest
+from craft_parts import Part, PartInfo, ProjectInfo
+from craft_parts.plugins.base import BasePythonPlugin, PluginProperties
+from overrides import override
+
+
+class AwesomePythonPluginProperties(PluginProperties, frozen=True):
+    plugin: Literal["awesomepy"] = "awesomepy"
+    source: str  # pyright: ignore[reportGeneralTypeIssues]
+
+
+class AwesomePythonPlugin(BasePythonPlugin):
```

Comment:
The class names are entertaining but `Fake` may be a more functional prefix as most test fixtures and helpers across the codebase use `Fake` or `Mock`.  I like that convention because I don't have to look at the implementation to know if it's mocking or faking data.

Reply:
This is a fake - it provides a simplified version of the implementation but as a real implementation.
```suggestion
class FakePythonPlugin(BasePythonPlugin):
```